packages/eventual-send/src/index.d.ts

```diff
@@ -97,79 +172,80 @@ declare interface HandledPromiseConstructor
   prototype: Promise<unknown>;
 }
 
-declare var HandledPromise: HandledPromiseConstructor;
-
 declare namespace global {
+  // eslint-disable-next-line vars-on-top,no-var
   var HandledPromise: HandledPromiseConstructor;
 }
 
-declare function makeHandledPromise(): HandledPromiseConstructor;
+export declare const HandledPromise: HandledPromiseConstructor;
+
+/**
+ * "E" short for "Eventual", what we call something that has to return a promise.
+ */
+type ECallable<T extends Callable> = ReturnType<T> extends PromiseLike<infer U> // function already returns a promise
+  ? T // make it return a promise
+  : (...args: Parameters<T>) => Promise<Awaited<ReturnType<T>>>;
 
 /* Types for E proxy calls. */
-type ESingleMethod<T> = {
-  readonly [P in keyof T]: (
-    ...args: Parameters<T[P]>
-  ) => Promise<Awaited<ReturnType<T[P]>>>;
+
+/**
+ * Ensure each function in T returns a promise
+ */
+type EMethods<T> = {
+  readonly [P in keyof T]: ECallable<T[P]>;
 };
-type ESingleCall<T> = T extends Function
-  ? ((...args: Parameters<T>) => Promise<Awaited<ReturnType<T>>>) &
-      ESingleMethod<Required<T>>
-  : ESingleMethod<Required<T>>;
-type ESingleGet<T> = {
-  readonly [P in keyof T]: Promise<Awaited<T[P]>>;
+
+type ECallableOrMethods<T> = T extends Callable
+  ? ECallable<T> & EMethods<Required<T>>
+  : EMethods<Required<T>>;
+
+type EGetters<T> = {
+  readonly [P in keyof T]: T[P] extends PromiseLike<infer U>
+    ? T[P]
+    : Promise<Awaited<T[P]>>;
```

Comment:
Is `Awaited` necessary here? The condition on 203 implies that `T[P]` is not PromiseLike. 

Though the fact that `Awaited` works here make me think we don't need the conditional:
```
readonly [P in keyof T]: T[P];
```
And in that case why do we need `EGetters` ? Perhaps it needs a `never` case for members that aren't getters.

Reply:
I need this construct for union types like `Foo | PromiseLike<Bar>` which combined are not `PromiseLike`.  Then the result is `Promise<Foo | Bar>`.  But the first `extends` is necessary so that I don't strip away template parameters by repackaging something that is already `PromiseLike`.